eng/tools/ToolDescriptionEvaluator/Program.cs

```diff
@@ -494,116 +536,83 @@ private static string FindToolDir(string repoRoot, string exeDir)
+                throw new FileNotFoundException($"Could not locate {serverExe} CLI artifact in Debug/Release outputs under servers.");
             }
+        }
 
-            var isDll = string.Equals(cliArtifact.Extension, ".dll", StringComparison.OrdinalIgnoreCase);
-            var fileName = isDll ? "dotnet" : cliArtifact.FullName;
-            var arguments = isDll ? $"{cliArtifact.FullName} tools list" : "tools list";
+        var isDll = string.Equals(cliArtifact.Extension, ".dll", StringComparison.OrdinalIgnoreCase);
+        var fileName = isDll ? "dotnet" : cliArtifact.FullName;
+        var arguments = isDll ? $"{cliArtifact.FullName} tools list" : "tools list";
 
-            var process = new Process
+        var process = new Process
+        {
+            StartInfo = new ProcessStartInfo
             {
-                StartInfo = new ProcessStartInfo
-                {
-                    FileName = fileName,
-                    Arguments = arguments,
-                    UseShellExecute = false,
-                    RedirectStandardOutput = true,
-                    RedirectStandardError = true,
-                    CreateNoWindow = true
-                }
-            };
-
-            process.Start();
-
-            var output = await process.StandardOutput.ReadToEndAsync();
-            var error = await process.StandardError.ReadToEndAsync();
-
-            await process.WaitForExitAsync();
+                FileName = fileName,
+                Arguments = arguments,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            }
+        };
 
-            if (process.ExitCode != 0)
-            {
-                if (isCiMode)
-                {
-                    return null; // Graceful fallback in CI
-                }
+        process.Start();
 
-                throw new InvalidOperationException($"Failed to get tools from azmcp: {error}");
-            }
+        var output = await process.StandardOutput.ReadToEndAsync();
+        var error = await process.StandardError.ReadToEndAsync();
 
-            // Filter out non-JSON lines (like launch settings messages)
-            var lines = output.Split('\n');
-            var jsonStartIndex = -1;
+        await process.WaitForExitAsync();
 
-            for (int i = 0; i < lines.Length; i++)
-            {
-                if (lines[i].Trim().StartsWith("{"))
-                {
-                    jsonStartIndex = i;
+        if (process.ExitCode != 0)
+        {
+            throw new IOException($"Failed to get tools from {fileName}: {error}");
+        }
 
-                    break;
-                }
-            }
+        // Filter out non-JSON lines (like launch settings messages)
+        var lines = output.Split('\n');
+        var jsonStartIndex = -1;
```

Comment:
Is this being done for reasons that `JsonSerializer.Deserialize` doesn't skip space and newline characters until seeing the first valid JSON start character?